src/todo_list.py

```diff
@@ -0,0 +1,309 @@
+            return True
+        else:
+            week_days_dict = { i : 0 for i in calendar.day_name }
+            for day_of_the_week in occurrences:
+                if day_of_the_week not in week_days_dict:
+                    return False
+            self.update_occurrence(None, occurrences)
+        return True
+
+    def validate_and_save_daily_occurrence(self, occurrences):
+        if len(occurrences) > 0:
+            return False
+        return True
+
+
+class Schedule(JSONRecord):
+    def __init__(self):
+        self.schedule_info = {
+            "cadence": "",  # once, yearly, monthly, weekly, daily
+            "occurrences": Occurrences(),
+            "start_date": None,
+            "end_date": None,
+            "due_date": None,
+        }
+        super().__init__(self.schedule_info)
+
+    def validate_occurrence(self, cadence, occurrences):
+        if cadence == "once":
+            if not self.schedule_info["occurrences"].validate_and_save_once_occurrence(occurrences):
+                raise Exception("Error: validate occurrence failed and didn't save for cadence type: once")
+        elif cadence == "yearly":
+            if not self.schedule_info["occurrences"].validate_and_save_yearly_occurrence(occurrences):
+                raise Exception("Error: validate occurrence failed and didn't save for cadence type: yearly")
+        elif cadence == "monthly":
+            if not self.schedule_info["occurrences"].validate_and_save_monthly_occurrence(occurrences):
+                raise Exception("Error: validate occurrence failed and didn't save for cadence type: monthly")
+        elif cadence == "weekly":
+            if not self.schedule_info["occurrences"].validate_and_save_weekly_occurrence(occurrences):
+                raise Exception("Error: validate occurrence failed and didn't save for cadence type: weekly")
+        elif cadence == "daily":
+            if not self.schedule_info["occurrences"].validate_and_save_daily_occurrence(occurrences):
+                raise Exception("Error: validate occurrence failed and didn't save for cadence type: daily")
+        else:
+            raise Exception("Error: cadence does not exist: {}".format(cadence))
+        return None
+
+    def update_schedule(self, cadence, occurrences):
+        err = self.validate_occurrence(cadence, occurrences)
+        if err is not None:
+            return None
+        # todo
+
+
+class Project(JSONRecord):
+    def __init__(self):
+        self.project_breakdown = {
+            "project": "",
+            "sub_project": "",
+            "parent_task": None
+        }
+        super().__init__(self.project_breakdown)
+    # todo
+
+class Task:
+    def __init__(self):
+        self.idx = None
+        self.task_name = None
+        self.created_on = None
+        self.schedule = Schedule()
+        self.is_complete = {}
+        self.task_dependencies = {}
+        self.project = Project()
```

Comment:
It seems like this is entangling how tasks depend on each other with how projects depend on each other. Would be clearer to have self.project and self.parent_task here, and keep track of project relationships in a different table